### ISSUE TITLE
Question: why are some features that seem to be complete are not implemented in any of the recent releases?

### ISSUE THREAD

**Issue by BillCombs** [Regalis11/Barotrauma#787]
Just want to clarify, i am not trying to pressure or demand you to implement certain features or do it sooner than other, especially if the other ones are more important. You are the developers, and it's up to you to decide how to plan your work. And, even after finishing making the feature, you are still free to change your mind, if you suddenly feel like the feature shouldn't be in the game.
But it just came to my mind that people have a tendency to forget some things even when they are really trying to keep them all in mind, and this could just possibly be the thing that happened in this case. There are not many sources of news about the development besides the blog, commits and changelog, so i thought that i would better clarify it here right now instead of just silently waiting for any changes forever, hoping that everything is fine and things will happen any time soon.

For example, f86aa1f would be pretty nice to have in the game right now, since it affects player communication, relationships between players, and, in some way, the whole player base. And it looks complete, at least judging by its title. But it seems like it has not been added yet(judging by not seeing it mentioned in any of the changelogs and by ingame testing), and i don't really understand why. Though i have nearly 0 experience in coding and game development, so i might not see the problems that might be preventing you from doing it. Perhaps the features are complete but are also unstable or not fully tested, so you decided to put them away for now?

**Comment by NilanthAnimosus** [comment]
Dev has made a lot of changes that don't even look similar anymore to the master branch. dev (and branches of dev) are so different that the code is not so simple to merge or copy paste. Although there are many features of dev we (me included) would want to have backported, it is typically not so simple to do.

In many cases, to have the smaller changes, it would mean having all the other changes or re-making said change entirely.

The entirety of dev to my understanding is essentially a mix of incomplete and complete, untested, tested, and in-development (both conceptually, design and how it works) features. 

Master is the more stable, less feature-testing version of the game we have.

**Comment by Regalis11** [comment]
The main reason why most of the new features in the dev branch haven't been added to master is that we want to have a decent amount of new content in the Steam version. If we kept releasing all the new features as they're added up until the Steam release, there would be less of an incentive to buy the game and the release would basically just be another patch with some additional Steam functionality. Also, just in case you haven't seen it yet, I made a [blog post](http://undertowgames.com/blog/barotrauma-status-update-steam-release/) recently about our current plans regarding Steam.

The point raised by Nilanth's is another major reason. There are some small fixes and additions that would be nice to have in the public version, but many of them are difficult, time-consuming and error-prone to merge to master because of the amount of changes in the dev version. For instance the commit you mentioned would not even compile if just cherry-picked to master.
